### PR TITLE
Add NaViT: Native Resolution Vision Transformer with Patch n' Pack

### PR DESCRIPTION
**Pull request by vikashg**
Adds NaViT (monai.networks.nets.NaViT), a Vision Transformer that removes the fixed-resolution constraint of standard ViT by packing multiple variable-size images into a single sequence per batch element.

Key features:
- Patch n' Pack: multiple images concatenated into one sequence per group, with a per-image attention mask preventing cross-image attention
- Factorised positional embeddings: separate learnable tables per spatial axis, allowing generalisation to unseen resolutions
- Token dropout: configurable fraction of patch tokens dropped during training (float or callable)
- Attention pooling: learned query attends over each image's tokens to produce a fixed-size per-image representation
- QK normalisation: RMS normalisation on queries and keys (ViT-22B style)
- 2D and 3D support: works for (C, H, W) and (C, H, W, D) inputs

Changes:
- monai/networks/nets/navit.py: new NaViT implementation
- monai/networks/nets/__init__.py: export NaViT
- tests/networks/nets/test_navit.py: 24 unit tests covering shape, variable resolutions, token dropout, auto-grouping, gradient flow, ill arguments, and forward validation
- docs/source/networks.rst: autoclass entry
- docs/source/whatsnew_1_5_2.md: feature description
- CHANGELOG.md: entry under Unreleased

Fixes # .

### Description

A few sentences describing the changes proposed in this pull request.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/networks/nets/test_navit.py [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+            "hidden_size": 128,
+            "mlp_dim": 256,
+            "num_layers": 2,
+            "num_heads": 4,
+            "in_channels": 1,
+            "spatial_dims": 2,
+        },
+        [[(1, 64, 64), (1, 32, 32), (1, 64, 32)]],
+        (3, 5),
+    ),
```
Comment (ericspod) [inline tests/networks/nets/test_navit.py]:
```suggestion
DEFAULT_KWARGS = {
    "image_size": 64,
    "patch_size": 16,
    "num_classes": 10,
    "hidden_size": 128,
    "mlp_dim": 256,
    "num_layers": 2,
    "num_heads": 4,
    "in_channels": 3,
    "spatial_dims": 2,
}
TEST_CASES_SHAPE = [
    # 2D single image
    (
        DEFAULT_KWARGS,
        [[(3, 64, 64)]],
        (1, 10),
    ),
    # 2D multiple images in one group
    (
        {   
            **DEFAULT_KWARGS,
            "num_classes": 5,
            "in_channels": 1,
            "spatial_dims": 2,
        },
        [[(1, 64, 64), (1, 32, 32), (1, 64, 32)]],
        (3, 5),
    ),
```
You can do something like this to reduce duplication in the test definition and in your tests below.

File: tests/networks/nets/test_navit.py [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+            hidden_size=128,
+            mlp_dim=256,
+            num_layers=2,
+            num_heads=4,
+            in_channels=1,
+            spatial_dims=2,
+        )
+        net.eval()
+        with self.assertRaises(ValueError):
+            net([[torch.randn(1, 50, 64)]])
```
Comment (ericspod) [inline tests/networks/nets/test_navit.py]: These tests could be merged with `parameterized` though that does lose the names and docstrings. The arguments for `NaViT` here could at least be put in a variable at the top to reduce total lines.

File: tests/networks/nets/test_navit.py (hunk 1 above)
Comment (vikashg) [reply tests/networks/nets/test_navit.py]: Done. Extracted DEFAULT_2D_KWARGS and DEFAULT_3D_KWARGS at the top and used {**DEFAULT_2D_KWARGS, ...} spread syntax throughout TEST_CASES_SHAPE and TEST_CASES_ILL_ARG.

File: tests/networks/nets/test_navit.py (hunk 2 above)
Comment (vikashg) [reply tests/networks/nets/test_navit.py]: Done. Merged the three individual test_forward_validation_* methods into a single @parameterized.expand test using a TEST_CASES_FORWARD_VALIDATION list. Net kwargs are shared via the default dict.

File: monai/networks/nets/navit.py [hunk 3]
```diff
@@ -0,0 +1,574 @@
...
+            x: input tensor of shape ``(B, N, C)``.
+            attn_mask: packed-sequence attention mask of shape ``(B, 1, N, N)``.
+
+        Returns:
+            Tensor of shape ``(B, N, C)``.
+        """
+        x = self.attn(x, attn_mask=attn_mask) + x
+        x = self.norm(x)
+        x = self.mlp(x) + x
+        return x
```
[the remark this answers is not in the document]
Comment (vikashg) [reply monai/networks/nets/navit.py:218]: Fixed. Changed to x = self.mlp(self.norm(x)) + x — the norm now only feeds the MLP while the raw residual is preserved. Good catch.

File: monai/networks/nets/navit.py [hunk 4]
```diff
@@ -0,0 +1,574 @@
...
+        super().__init__()
+
+        if spatial_dims not in (2, 3):
+            raise ValueError("spatial_dims must be 2 or 3.")
+        if not (0 <= dropout_rate <= 1):
+            raise ValueError("dropout_rate should be between 0 and 1.")
+        if not (0 <= emb_dropout_rate <= 1):
+            raise ValueError("emb_dropout_rate should be between 0 and 1.")
+        if hidden_size % num_heads != 0:
+            raise ValueError("hidden_size should be divisible by num_heads.")
```
[the remark this answers is not in the document]
Comment (vikashg) [reply monai/networks/nets/navit.py:323]: Fixed. Added if num_heads <= 0: raise ValueError(...) before the divisibility check.

File: monai/networks/nets/navit.py [hunk 5]
```diff
@@ -0,0 +1,574 @@
...
+        patch_dim = in_channels * (patch_size**spatial_dims)
+        self.to_patch_embedding = nn.Sequential(
+            nn.LayerNorm(patch_dim), nn.Linear(patch_dim, hidden_size), nn.LayerNorm(hidden_size)
+        )
+
+        # --- factorised positional embeddings (one table per spatial axis) ---
+        image_size_t = ensure_tuple_rep(image_size, spatial_dims)
+        self.pos_embed_axes = nn.ParameterList(
+            [nn.Parameter(torch.randn(img_d // patch_size, hidden_size)) for img_d in image_size_t]
+        )
```
[the remark this answers is not in the document]
Comment (vikashg) [reply monai/networks/nets/navit.py]: Fixed. Constructor now validates each dimension of image_size is divisible by patch_size and raises ValueError with a clear message if not.

File: tests/networks/nets/test_navit.py [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+        """Token dropout should not be applied during eval, producing deterministic output."""
+        net = NaViT(
+            **{**DEFAULT_2D_KWARGS, "num_classes": 5, "in_channels": 1},
+            token_dropout_prob=0.5,
+        )
+        net.eval()
+        input_data = [[torch.randn(1, 64, 64)]]
+        out1 = net(input_data)
+        out2 = net(input_data)
+        self.assertTrue(torch.allclose(out1, out2))
```
[the remark this answers is not in the document]
Comment (vikashg) [reply tests/networks/nets/test_navit.py]: done

**Review by ericspod — APPROVED** [review]
Looks good to me, thanks!
